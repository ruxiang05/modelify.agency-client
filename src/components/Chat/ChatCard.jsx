import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import '../../styles/chat-card.scss';
import { ReactComponent as UnknownUserIcon } from '../../assets/icons/user-circle.svg';
import { ReactComponent as GoTo } from '../../assets/icons/arrow-right.svg';

const ChatCard = ({ chat }) => (
  <Link
    to={{
      pathname: `/chats/${chat._id}`,
      state: {
        chat,
      },
    }}
  >
    <div className="chat-card">
      <div className="content">
        <UnknownUserIcon />
        <div>{chat.job.title}</div>

      </div>
      <GoTo />
    </div>
  </Link>
);

ChatCard.propTypes = {
  chat: PropTypes.shape({}).isRequired,
};
export default ChatCard;
