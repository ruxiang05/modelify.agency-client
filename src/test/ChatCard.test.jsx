/* Uses methods and/or components from react and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import ChatCard from '../components/Chat/ChatCard';

const chat = {
  id: '1',
  job: {
    title: 'title',
  },
};

describe('ChatCard', () => {
  it('should render ChatCard', () => {
    const wrapper = shallow(<ChatCard chat={chat} />);
    expect(wrapper).toMatchSnapshot();
  });
});
