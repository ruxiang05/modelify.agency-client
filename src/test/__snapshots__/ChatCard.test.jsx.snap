// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatCard should render ChatCard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatCard
    chat={
      Object {
        "id": "1",
        "job": Object {
          "title": "title",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="chat-card"
      >
        <div
          className="content"
        >
          <ForwardRef />
          <div>
            title
          </div>
        </div>
        <ForwardRef />
      </div>,
      "replace": false,
      "to": Object {
        "pathname": "/chats/undefined",
        "state": Object {
          "chat": Object {
            "id": "1",
            "job": Object {
              "title": "title",
            },
          },
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="content"
          >
            <ForwardRef />
            <div>
              title
            </div>
          </div>,
          <ForwardRef />,
        ],
        "className": "chat-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef />,
              <div>
                title
              </div>,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "title",
              },
              "ref": null,
              "rendered": "title",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="chat-card"
        >
          <div
            className="content"
          >
            <ForwardRef />
            <div>
              title
            </div>
          </div>
          <ForwardRef />
        </div>,
        "replace": false,
        "to": Object {
          "pathname": "/chats/undefined",
          "state": Object {
            "chat": Object {
              "id": "1",
              "job": Object {
                "title": "title",
              },
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="content"
            >
              <ForwardRef />
              <div>
                title
              </div>
            </div>,
            <ForwardRef />,
          ],
          "className": "chat-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ForwardRef />,
                <div>
                  title
                </div>,
              ],
              "className": "content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "title",
                },
                "ref": null,
                "rendered": "title",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
